src/abstractdataframe/iteration.jl

```diff
@@ -176,14 +176,23 @@ Base.IteratorSize(::Type{<:DataFrameColumns}) = Base.HasShape{1}()
 Base.size(itr::DataFrameColumns) = (size(parent(itr), 2),)
 
 function Base.size(itr::DataFrameColumns, d::Integer)
-    d < 1 && throw(ArgumentError("dimension out of range"))
-    return d == 1 ? size(itr)[1] : 1
+    d != 1 && throw(ArgumentError("dimension out of range"))
+    return size(itr)[1]
 end
 
+Base.ndims(::DataFrameColumns) = 1
+Base.ndims(::Type{<:DataFrameColumns}) = 1
+
 Base.length(itr::DataFrameColumns) = size(itr)[1]
 Base.eltype(::Type{<:DataFrameColumns}) = AbstractVector
+
 Base.firstindex(itr::DataFrameColumns) = 1
 Base.lastindex(itr::DataFrameColumns) = length(itr)
+
+Base.firstindex(itr::DataFrameColumns, i::Integer) = first(axes(itr, i))
+Base.lastindex(itr::DataFrameColumns, i::Integer) = last(axes(itr, i))
+Base.axes(itr::DataFrameColumns, i::Integer) = Base.OneTo(size(itr, i))
```

Comment:
Do we need these in the end given that a fallback was added to Base?

Reply:
this fallback is only in 1.6. I added it to have it also e.g. in 1.0 in case someone in the future uses these functions in user code explicitly.

Reply:
Maybe add a `VERSION` check? It's always nice to be able to remove code at some point in the future. :-)

Reply:
added - checking if I did it correctly is welcome ;). Thank you!

Reply:
@nalimilan - does the version check look good to you now?